test/solidity/Facets/DexManagerFacet.t.sol

```diff
@@ -109,29 +131,161 @@ contract DexManagerFacetTest is DSTest, DiamondTest {
                 ++i;
             }
         }
+
+        vm.stopPrank();
     }
 
-    function testFailAddZeroAddress() public {
+    function testFail_AddZeroAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
         dexMgr.addDex(address(0));
+
+        vm.stopPrank();
     }
 
-    function testFailAddNonContract() public {
+    function testFail_AddNonContract() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
         dexMgr.addDex(address(1337));
+
+        vm.stopPrank();
     }
 
-    function testFailBatchAddZeroAddress() public {
+    function testFail_BatchAddZeroAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
         address[] memory dexs = new address[](3);
         dexs[0] = address(c1);
         dexs[1] = address(c2);
         dexs[2] = address(0);
         dexMgr.batchAddDex(dexs);
+
+        vm.stopPrank();
     }
 
-    function testFailBatchAddNonContract() public {
+    function testFail_BatchAddNonContract() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
         address[] memory dexs = new address[](3);
         dexs[0] = address(c1);
         dexs[1] = address(c2);
         dexs[2] = address(1337);
         dexMgr.batchAddDex(dexs);
+
+        vm.stopPrank();
+    }
+
+    function testRevert_addDex_NotOwner() public {
+        vm.expectRevert(UnAuthorized.selector);
+
+        vm.prank(address(0xdead)); // prank a non-owner to attempt adding a DEX
```

Comment:
I would swap this line with expectRevert

same below...prank in the beginning 
Before you argued that expectRevert and the next call belong closely together....here there is something in between...